benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/JavaClientBenchmarkRunner.java

```diff
@@ -0,0 +1,257 @@
+package com.google.cloud.pgadapter.tpcc;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.pgadapter.tpcc.config.PGAdapterConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.SpannerConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.TpccConfiguration;
+import com.google.cloud.spanner.AbortedException;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.SessionPoolOptions;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import com.google.cloud.spanner.TransactionManager;
+import com.google.cloud.spanner.Value;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.Uninterruptibles;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class JavaClientBenchmarkRunner extends AbstractBenchmarkRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(JavaClientBenchmarkRunner.class);
+  private final ThreadLocal<Spanner> spannerThreadLocal = new ThreadLocal<>();
+  private final ThreadLocal<DatabaseClient> databaseClientThreadLocal = new ThreadLocal<>();
+  private final ThreadLocal<TransactionManager> transactionManagerThreadLocal = new ThreadLocal<>();
+  private final ThreadLocal<TransactionContext> transactionThreadLocal = new ThreadLocal<>();
+
+  JavaClientBenchmarkRunner(
+      Statistics statistics,
+      TpccConfiguration tpccConfiguration,
+      PGAdapterConfiguration pgAdapterConfiguration,
+      SpannerConfiguration spannerConfiguration,
+      Metrics metrics) {
+    super(statistics, tpccConfiguration, pgAdapterConfiguration, spannerConfiguration, metrics);
+  }
+
+  void setup() throws SQLException, IOException {
+    Spanner spanner = createSpanner();
+    spannerThreadLocal.set(spanner);
+    databaseClientThreadLocal.set(
+        spanner.getDatabaseClient(
+            DatabaseId.of(
+                spannerConfiguration.getProject(),
+                spannerConfiguration.getInstance(),
+                spannerConfiguration.getDatabase())));
+    transactionManagerThreadLocal.set(databaseClientThreadLocal.get().transactionManager());
+    Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(2000L));
+  }
+
+  void teardown() throws SQLException {
+    transactionManagerThreadLocal.get().close();
+    spannerThreadLocal.get().close();
+  }
+
+  void executeStatement(String sql) throws SQLException {
+    if (sql.equals("begin transaction")) {
+      transactionThreadLocal.set(transactionManagerThreadLocal.get().begin());
```

Comment:
Does this really work? As in: `TransactionManager` instances are not intended to execute multiple transactions. Executing `TransactionManager#begin()` multiple times using the same `TransactionManager` should normally not work.

Reply:
Actually, it works. I thought we can call `TransactionManager` to begin a new transaction and then call `transactionManagerThreadLocal.get().commit();` or `transactionManagerThreadLocal.get().rollback();` to complete the transaction. After that, we can call `begin` again to get a new transaction. 

From what you say, do we always need to create a new `TransactionManager` to begin a new transaction?

Reply:
Fixed.